eng/bomgenerator/src/main/java/com/azure/tools/bomgenerator/BomGenerator.java

```diff
@@ -338,8 +358,18 @@ private void writeBom(Collection<BomDependency> bomDependencies) {
         dependencies.sort(new DependencyComparator());
 
         // Remove external dependencies from the BOM.
-        dependencies = dependencies.stream().filter(dependency -> BASE_AZURE_GROUPID.equals(dependency.getGroupId())).collect(Collectors.toList());
+        Set<String> premiumLibraries = getPremiumLibraries();
+        dependencies = dependencies.stream().filter(dependency -> groupIds.contains(dependency.getGroupId())).collect(Collectors.toList());
         management.setDependencies(dependencies);
         writeModel(this.pomFileName, this.outputFileName, model);
     }
+
+    private Set<String> getPremiumLibraries() {
+        return Set.of("azure-resourcemanager", "azure-resourcemanager-appplatform", "azure-resourcemanager-appservice", "azure-resourcemanager-authorization",
+                "azure-resourcemanager-cdn", "azure-resourcemanager-compute", "azure-resourcemanager-containerinstance",
+                "azure-resourcemanager-containerregistry", "azure-resourcemanager-containerservice", "azure-resourcemanager-cosmos",
+                "azure-resourcemanager-dns", "azure-resourcemanager-eventhubs", "azure-resourcemanager-keyvault", "azure-resourcemanager-monitor"
+                , "azure-resourcemanager-msi", "azure-resourcemanagernetwork", "azure-resourcemanager-privatedns", "azure-resourcemanager-redis", "azure-resourcemanager-resources"
+                , "azure-resourcemanager-search", "azure-resourcemanager-servicebus", "azure-resourcemanager-sql", "azure-resourcemanager-storage", "azure-resourcemanager-trafficmanager");
+    }
```

Comment:
One alternative is to only include those from `patch_release_client.txt` for `com.azure.resourcemanager` groupId. 
Though still facing following issues:
1. Current BOM contains more packages than `patch_release_client.txt`, except for core-related libraries. Thus the filter may only apply to `com.azure.resourcemanager` groupId.
2. In core-v2, groupId for ARM libraries will change, and we'll still need an adjustment then.

Reply:
Changed to ArtifactFilter:
https://github.com/Azure/azure-sdk-for-java/pull/45655#discussion_r2169506729